src/views/datasources/list/CreateDataSourceModal/CreateDataSourceModal.tsx

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
```

Comment:
I would suggest to breakdown to small components to have better readability of this form, also create constants for hard-coded strings/numbers that you are using more than once